libcontainer/system/proc.go

```diff
@@ -47,7 +49,7 @@ func (s State) String() string {
 // Stat_t represents the information from /proc/[pid]/stat, as
 // described in proc(5) with names based on the /proc/[pid]/status
 // fields.
-type Stat_t struct {
+type Stat_t struct { //nolint:revive // ignore "type Stat_t should be named StatT"
```

Comment:
Do we need to export this type at all? Looks like we don't. I have renamed it to `statT` and everything works.

Reply:
The type is returned by an exported function; it's good practice to export the type if it's returned by an exported function.

I assumed we chose `Stat_t` as name here to match `syscall.Stat_t` / `unix.Stat_t`, but we could rename it to `StatT` to use Go conventions; looks like that's what we did in Moby

Reply:
Renamed to `StatT`